src/MCMC.jl

```diff
@@ -173,24 +172,16 @@ end
 function proposal(mcmc::MCMCObj)
 
     variances = zeros(length(mcmc.param))
-    for (idx, prior) in enumerate(mcmc.prior)
+    priors = [mcmc.prior[i].dist for i in 1:length(mcmc.prior)]
+    for (idx, prior) in enumerate(priors)
         variances[idx] = var(prior)
     end
 
     if mcmc.algtype == "rwm"
-        #prop_dist = MvNormal(mcmc.posterior[1 + mcmc.iter[1], :], (mcmc.step[1]) * Diagonal(variances))
         prop_dist = MvNormal(zeros(length(mcmc.param)), (mcmc.step[1]^2) * Diagonal(variances))
     end
     sample = mcmc.posterior[1 + mcmc.iter[1], :] .+ rand(prop_dist)
 
```

Comment:
Did you want to put in some kind of inbounds check? perhaps with just a message if the proposal leaves the support of the prior - something like - " MCMC proposal is leaving the support of the prior for parameter i, consider using a transform or a prior with wider support" ? .

remark on this is that I have switch to using Normal priors under a pushforward through a sigmoid function as opposed to using a uniform prior etc. 

Then resample until it is within bounds.

Reply:
@odunbar: Sounds like a reasonable thing to do - I'll implement this!